### ISSUE TITLE
Replace `django-rest-swagger` with `drf-spectacular`

### ISSUE THREAD

**Issue by afuetterer** [rdmorganiser/rdmo#698]
### Rationale / Begründung

`django-rest-swagger` is deprecated since 2019. Should rdmo switch to an alternative openapi library? `drf-spectacular` looks promising.

### Affected 

Software group

### References / Verweise

* https://github.com/marcgibbons/django-rest-swagger
* https://github.com/tfranzel/drf-spectacular

**Comment by afuetterer** [comment]
@jochenklar @MyPyDavid: Do you know, if the swagger UI used or is this just for convenience?

I want to switch out the deprecated library, that generates 100+ warnings when running the tests.

Is it important to generate the exact same UI under the exact same URL?

**Comment by jochenklar** [comment]
Hi @afuetterer thanks for the effort! No I don't think we need swagger, I never liked it anyway. But I guess we need something like this, meaning a browsable API explorer thingy.

**Comment by afuetterer** [comment]
Alright, thanks.

drf-spectacular generates the openapi schema in json/yaml and renders those in swagger/redoc.

I will submit a PR, then you can test it.

**Comment by afuetterer** [comment]
Regarding the versioning here:

There is the RDMO version, e.g. `2.1.3` or `2.2.0`, and there is a URL path versioning `api/v1/`.

What is the difference here?

Is the "version" of the API always "v1"?

**Comment by jochenklar** [comment]
Yes, the version is aleays v1, and its there in case we have another one and need to run them both in parallel. A lot of people say that it is super important to version apis in this way, so I try to do it all the time. An I needed it *once*.

**Comment by afuetterer** [comment]
```console
$ python manage.py spectacular --color --file schema.yml --validate
[...]
Schema generation summary:
Warnings: 237 (109 unique)
Errors:   28 (5 unique)
```

**Comment by afuetterer** [comment]
This is the swagger schema generated by django-rest-swagger:
[django-rest-swagger.zip](https://github.com/user-attachments/files/16201224/django-rest-swagger.zip)
(I could not upload a yaml file.)

```console
npx open-swagger-ui --open django-rest-swagger.yaml
```

**Comment by afuetterer** [comment]
What about this repo?

https://github.com/rdmorganiser/rdmo-openapi

**Comment by jochenklar** [comment]
I think https://github.com/rdmorganiser/rdmo-openapi was used by @triole a long time ago. If I understood correctly, we need to include the schema into the `rdmo` repos anyway if we want to use `drf-spectacular`, right?

**Comment by afuetterer** [comment]
Can we please archive https://github.com/rdmorganiser/rdmo-openapi then and add a reference, that this is out of date?

**Comment by jochenklar** [comment]
Ok so it is like with swagger before, fine for me.

**Comment by afuetterer** [comment]
It spits out a billion warnings though. Mostly about typing. But to be honest I am not in the mood to fix those. Let me open and rebase the PR again and you can look at it.

django rest swagger is deprecated and should go in my opinion.

Is the swagger view used a lot? Maybe this could be optional, in case an instance will not use it?

Anyway this stops us from supporting Python 3.13, which is bad.

**Comment by jochenklar** [comment]
Thanks for all the work, the `drf-spectacular` integration will be an optional feature in RDMO 2.3.0. (optional dependency: `pip install rdmo[openapi]`.
